src/main/java/com/tira/restaurants/validator/EmailValidator.java

```diff
@@ -0,0 +1,29 @@
+package com.tira.restaurants.validator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+
+public class EmailValidator implements ConstraintValidator<ValidEmail, String> {
+    private Pattern pattern;
+    private Matcher matcher;
+    private static final String EMAIL_PATTERN = "^[_A-Za-z0-9-\\+]+(\\.[_A-Za-z0-9-]+)*@" + "[A-Za-z0-9-]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,})$";
+
+    @Override
+    public void initialize(final ValidEmail constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(final String username, final ConstraintValidatorContext context) {
+        return (validateEmail(username));
+    }
+
+    private boolean validateEmail(final String email) {
+        pattern = Pattern.compile(EMAIL_PATTERN);
```

Comment:
If you decide to use your own regexp, you can move the compiled Pattern to a static final instead of compiling it every time you need it.